include/common/aglResCommon.h

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include <basis/seadRawPrint.h>
+#include <prim/seadBitUtil.h>
+
+namespace agl {
+
+// maybe first parameter is is_le, maybe it is big_endian - different between decomps
+void ModifyEndianU32(bool is_le, void* p_data, size_t size);
```

Comment:
Maybe the first parameter should be called "swap" instead? That'd explain why it's different between Wii U and Switch

```suggestion
void ModifyEndianU32(bool swap, void* p_data, size_t size);
```

Reply:
No. The implementation of the function seems to be the same:
https://github.com/aboood40091/sead/blob/master/packages/agl/src/utility/common/aglResCommon.cpp
https://github.com/MonsterDruide1/agl/blob/res-shader-stuff/src/utility/aglResCommon.cpp

What I'm more surprised at is that @MonsterDruide1 forgot to update/move source CPP files. This PR seems to only cover headers.

Reply:
I knowingly "ignored" the CPP for now, as this PR was done in an effort of gathering more information with headers, completely ignoring decomps/source files at the moment. If at all, I would do those later, and in a separate PR.

What about the suggesion from Leo above - should that be changed, or does the "no" mean that it is correct as it is right now?

Reply:
My point was that it definitely should be changed back to "is_le", because that's really what the parameter means.

Reply:
Well, it is `is_le` at the moment, as I haven't applied Leo's suggestion yet - so you mean it should stay like this.

Reply:
Yes.

By the way, I still think source CPP files should still be updated and moved in the same PR.

Reply:
Alright, I don't have a strong opinion on it - this means the PR will probably double in size, but I'm fine with that too. I'll start working on it.

Reply:
Done - I think. At least I only found a single source file in your repo, which majorly consists of mismatching functions for the switch though...

Reply:
You don't have to add that if it doesn't match. All I meant for you was to update existing files (so that they still compile with the new headers) and to rename files accordingly.

P.S. I think the 4th shader type is compute.

Reply:
I made sure that it compiles at all points, and already adjusted the source files accordingly - I still think that the additional information in non-matching functions can be helpful when working on related stuff.

Alright, renamed the shader type accordingly.